data.js

```diff
@@ -0,0 +1,51 @@
+const popura = require('popura');
+ 
+class AbstractAnimeDataSource {
+    getAnime(name) {
+        throw "Implementation error";
+    }
+
+    _constructAnimeModel(animeResponse) {
+        throw "Implementation error";
+    }
+}
+
+class Anime {
+    constructor(name, score) {
+        this.name = name;
+        this.score = score;
+    }
+}
+
+class MALClient extends AbstractAnimeDataSource {
+    constructor(popuraClient) {
+        super();
+        this._client = popuraClient;
+    }
+
+    getAnime(name) {
+        return new Promise((resolve, reject) => {
+            this._client.searchAnimes(name)
+                .then(response => {
+                    const result = response.shift();
+                    resolve(result === null ? result : this._constructAnimeModel(result));
+                })
+                .catch(err => reject(err));
+        });
+    }
+
+    _constructAnimeModel(animeResponse) {
+        try {
+            const title = animeResponse.title;
```

Comment:
Is there a better way to checking the `animeResponse` has these properties on it? HasOwnProperty??